pkg/apis/nfs.rook.io/v1alpha1/types.go

```diff
@@ -0,0 +1,102 @@
+/*
+Copyright 2018 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package v1alpha1
+
+import (
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ***************************************************************************
+// IMPORTANT FOR CODE GENERATION
+// If the types in this file are updated, you will need to run
+// `make codegen` to generate the new types under the client/clientset folder.
+// ***************************************************************************
+
+// +genclient
+// +genclient:noStatus
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type NetworkFileSystem struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+	Spec              NetworkFileSystemSpec `json:"spec"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type NetworkFileSystemList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+	Items           []NetworkFileSystem `json:"items"`
+}
+
+// NetworkFileSystemSpec represents the spec of NFS daemon
+type NetworkFileSystemSpec struct {
+	// Replicas of the NFS daemon
+	Replicas int `json:"replicas,omitempty"`
+
+	// The parameters to configure the NFS export
+	Exports []ExportsSpec `json:"exports,omitempty"`
+}
+
+// ExportsSpec represents the spec of NFS exports
+type ExportsSpec struct {
+	// Name of the export
+	Name string `json:"name,omitempty"`
+
+	// The NFS server configuration
+	Server []ServerSpec `json:"server,omitempty"`
+
+	// PVC from which the NFS daemon gets storage for sharing
+	PersistantVolumeClaim v1.PersistentVolumeClaimSpec `json:"name,omitempty"`
```

Comment:
Did you mean to have two fields in this struct have the same field in the yaml? `json:"name,omitempty"`?

Reply:
If so, I'm not sure how it works. It's also confusing- consider renaming it.

Reply:
Sorry about this one, I copied and pasted `json:*` part and forgot to edit this field.